### PR TITLE
baselayout: introduce /etc/profile.d/coreos-profile.sh

### PR DESCRIPTION
**Pull request by vcaputo**
This restores the update strategy and failed units output previously
included in the motd.

### CODE REVIEW

File: baselayout/home-coreos-dot-bash_profile [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+# runs your .bashrc and is recommended by the bash info pages.
+[[ -f ~/.bashrc ]] && . ~/.bashrc
+
+if ! systemctl is-active locksmithd > /dev/null; then
+	echo -e "Update Strategy: \033[31mNo Reboots\033[39m"
+fi
+
+FAILED=$(systemctl list-units --state=failed --no-legend)
+if [[ ! -z "${FAILED}" ]]; then
+	echo -e "Failed Units: \033[31m${count}\033[39m"
```
Comment (crawford) [inline baselayout/home-coreos-dot-bash_profile]: Where does `count` come from?

**Comment by marineam** [comment]
An impact of this change is that the failed units won't be listed in systems that are upgraded as we don't clobber .bash_profile if the link or file exists, including if it is a link to the original skel file.

Also .bash_profile is going to get sourced by non-interactive logins like scp right? If so it should skip doing anything in that case. Re-using the example check in .bashrc should work.

Maybe instead of changing the user dot file we add a script to /usr/share/profile.d/ or update /build/amd64-usr/etc/profile and check for the sudo or wheel groups or the core user. That would maintain behavior on GCE where the norm is to have per-user accounts instead of using the common core account.

File: baselayout/home-coreos-dot-bash_profile (hunk 1 above)
Comment (vcaputo) [inline baselayout/home-coreos-dot-bash_profile]: hah, good catch, that bug was already there!

**Comment by vcaputo** [comment]
All good points, a bit more thought is clearly needed before landing this.

**Comment by vcaputo** [comment]
@marineam well, the current form of this PR sticks it in /etc/profile.d/coreos-profile.sh and doesn't discriminate according to user or group since that aspect *feels* a bit janky to me, to encode such policy in an otherwise generic profile.d entity.

I'm leaning towards leaving it like this, and the end-user can choose to discard it from /etc/profile.d and source it from select .bash_profiles or a tweaked /etc/profile.d entry if they wish to restrict it to a specific user/group.   This PR restores the current behavior of showing this to any user out of the box as if it were the motd.  Thoughts?

**Comment by marineam** [comment]
Sounds reasonable, LGTM

**Pull request merged**